### PR TITLE
chore(backport): set python3.11 as the minimal supported version (#649)

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
Make python3.11 the minimal supported version of python, required for incoming features and dependencies.
Jira: https://issues.redhat.com/browse/AAP-19622

backport of https://github.com/ansible/ansible-rulebook/pull/649

### CODE REVIEW

**Review by ttuffin — APPROVED** [review]

**Comment by ttuffin** [comment]
@Alex-Izquierdo should this be targeting stable/1.0?

**Comment by Alex-Izquierdo** [comment]
> @Alex-Izquierdo should this be targeting stable/1.0?

Nope, is a breaking change. this is just to remove "devel" branch since main is currently the base for the future 1.1

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM quay.io/centos/centos:stream9-development
+FROM quay.io/centos/centos:stream9
```
[the remark this answers is not in the document]
Comment (ttuffin) [reply Dockerfile:1]: that's right :+1: thanks for the reminder.

**Comment by Alex-Izquierdo** [comment]
Solved conflicts, please review again

**Review by ttuffin — APPROVED** [review]

**Pull request closed without merging**